lib/src/platform.dart

```diff
@@ -0,0 +1,129 @@
+import 'pubspec.dart';
+
+abstract class KnownPlatforms {
+  static const String browser = 'browser';
+  static const String console = 'console';
+  static const String flutter = 'flutter';
+  static const String mirrors = 'mirrors';
+
+  static const String angular = 'angular';
+}
+
+class PlatformSummary {
+  final Platform package;
+  final Map<String, Platform> libraries;
+
+  PlatformSummary(this.package, this.libraries);
+
+  bool get hasConflict =>
+      package.hasConflict ||
+      libraries.values.any((p) => p.hasConflict) ||
+      libraries.values.any((p) => p.excludes
+          .any((s) => package.only.contains(s) || package.uses.contains(s)));
+}
+
+class Platform {
+  final List<String> only;
```

Comment:
I'd love to understand your logic here.

using `dart:html` should exclude flutter (like using `dart:mirrors` excludes flutter`).

I guess I'd expect just a `uses` field – and then a function that can answer "will this work on platform X" with the logic centralized.